src/metatrain/utils/data/dataset.py

```diff
@@ -62,20 +82,25 @@ def atomic_types(self) -> List[int]:
 
     @atomic_types.setter
     def atomic_types(self, value: List[int]):
-        self._atomic_types = set(value)
+        self._atomic_types = _set(value)
+
+    def to(self, device: torch.device) -> "DatasetInfo":
+        """Return a copy of this instance with all tensors moved to the specified device."""
+        new = self.copy()
+        for key, target_info in new.targets.items():
+            new.targets[key] = target_info.to(device)
+        for key, extra_data in new.extra_data.items():
+            new.extra_data[key] = extra_data.to(device)
+        return new
 
     def __repr__(self):
-        return (
-            f"DatasetInfo(length_unit={self.length_unit!r}, "
-            f"atomic_types={self.atomic_types!r}, targets={self.targets!r})"
+        return "DatasetInfo(length_unit={!r}, atomic_types={!r}, targets={!r})".format(
+            self.length_unit, self.atomic_types, self.targets
         )
 
     def __eq__(self, other):
         if not isinstance(other, DatasetInfo):
-            raise NotImplementedError(
-                "Comparison between a DatasetInfo instance and a "
-                f"{type(other).__name__} instance is not implemented."
-            )
+            return False
```

Comment:
Who introdcued this error raise here.... Return `False` should be always here. Thanks!